### PR TITLE
HPCC-23164 Ensure read/write stats updated when streaming

### PR DESCRIPTION
**Pull request by jakesmith**
When handling row stream, ensure read and write bytes are
tracked.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: fs/dafsserver/dafsserver.cpp [hunk 1]
```diff
@@ -1064,6 +1068,7 @@ class CRemoteRequest : public CSimpleInterfaceOf<IInterface>
             processRead(requestTree, responseMb);
         else if (activity->queryIsWriteActivity())
             processWrite(requestTree, restMb, responseMb);
+        activity->flushStatistics(stats);
```
Comment (mckellyln) [inline fs/dafsserver/dafsserver.cpp]: Who updates the bytesRead and/or bytesWritten for this ?

**Comment by mckellyln** [comment]
@jakesmith one ques

File: fs/dafsserver/dafsserver.cpp (hunk 1 above)
Comment (jakesmith) [reply fs/dafsserver/dafsserver.cpp]: depends on the activity implementation. For read bytesRead is updated via the IFileSerialStreamCallback callback,
For write, bytesWritten it's simpler, updated as write is called.

**Comment by jakesmith** [comment]
@mckellyln - see reply above.

File: fs/dafsserver/dafsserver.cpp (hunk 1 above)
Comment (mckellyln) [reply fs/dafsserver/dafsserver.cpp]: ok.
```
class CRemoteWriteBaseActivity : 
    virtual void write(size32_t sz, const void *rowData) override
    {
        throwUnexpected(); // method should be implemented in derived classes.
    }
    virtual void flushStatistics(CClientStats &stats) override
    {
        // NB: will be called by same thread that is writing.
        stats.addWrite(bytesWritten);
        bytesWritten = 0;
    }
```
And
```
class CRemoteDiskWriteActivity :
    virtual void write(size32_t sz, const void *rowData) override
    {
        checkOpen();
        iFileIOStream->write(sz, rowData);
        bytesWritten += sz;
    }
```

Comment (mckellyln) [reply fs/dafsserver/dafsserver.cpp]:
```
class CRemoteDiskBaseActivity : 
    virtual void flushStatistics(CClientStats &stats) override
    {
        throwUnexpected();
    }
```

Comment (mckellyln) [reply fs/dafsserver/dafsserver.cpp]: ok.  I guess I was originally looking for a base flushStatistics() that updated both reads and writes.

**Comment by mckellyln** [comment]
@jakesmith looks good.

**Review by mckellyln — APPROVED** [review]
Approved for merge.

**Comment by jakesmith** [comment]
@richardkchapman - please merge.

**Pull request merged**